vispy/scene/cameras/base_camera.py

```diff
@@ -465,13 +489,86 @@ def _set_scene_transform(self, tr):
-            try:
-                cam._linked_cameras_no_update = self
-                cam.set_state(self.get_state())
-            finally:
-                cam._linked_cameras_no_update = None
+        
+        if len(self._linked_cameras) != 0:
+            if isinstance(self._linked_cameras[0], list):
+                ## link only one axis between to cameras
+                if(self._linked_cameras[0][1] == "x"):
+                    for i in range(len(self._linked_cameras)):
+                        cam = self._linked_cameras[i][0]
+                        self_state = self.get_state()
+                        cam_state = cam.get_state()
+                        s = self_state
+                        for u in range(2):
+                            if(u == 1):
+                                if(self.last_self_pos is not None and self.last_cam_pos is not None and self.last_cam_top is not None and self.last_cam_bottom is not None  and self.last_self_top is not None and self.last_self_bottom is not None):
+                                    self_state["rect"].pos = (self_state["rect"].pos[0], self.last_cam_pos[1])
+                                    self_state["rect"].top = self.last_cam_top
+                                    self_state["rect"].bottom = self.last_cam_bottom
+                                    cam.set_state(self_state)
+                                    self_state["rect"].pos = (self_state["rect"].pos[0], self.last_self_pos[1])
+                                    self_state["rect"].top = self.last_self_top
+                                    self_state["rect"].bottom = self.last_self_bottom
+                                    self.set_state(self_state)
+                                    self.last_self_pos = self.get_state()["rect"].pos
+                                    self.last_cam_pos = cam.get_state()["rect"].pos
+                                else:
+                                    self.last_self_pos = self.get_state()["rect"].pos
+                                    self.cam_self_pos = cam.get_state()["rect"].pos
+                                    self.last_self_top = self.get_state()["rect"].top
+                                    self.last_cam_top = cam.get_state()["rect"].top
+                                    self.last_self_bottom = self.get_state()["rect"].bottom
+                                    self.last_cam_bottom = cam.get_state()["rect"].bottom
+                            else:
+                                self.last_self_pos = self.get_state()["rect"].pos
+                                self.last_cam_pos = cam.get_state()["rect"].pos
+                                self.last_self_top = self.get_state()["rect"].top
+                                self.last_cam_top = cam.get_state()["rect"].top
+                                self.last_self_bottom = self.get_state()["rect"].bottom
+                                self.last_cam_bottom = cam.get_state()["rect"].bottom
```

Comment:
Can the `u == 0` and `u == 1` cases be put in to helper methods/functions to clean up this code and hopefully reuse it between x and y?

Reply:
Well i should probably create a save_old_rect method however i am not quite shore how to create a method which is reusable between x and y since different variables have to be changed without the same if x elif y else structure